api/src/main/java/org/openmrs/module/patientflags/api/impl/FlagServiceImpl.java

```diff
@@ -224,7 +233,19 @@ public void purgeFlag(Integer flagId) {
 		// then refresh the cache
 		refreshCache();
 	}
-	
+
+	public void retireFlag(Flag flag, String retiredReason) {
+		if (retiredReason != null && retiredReason.length() >= 1) {
+			flag.setRetired(true);

```

Comment:
Shouldn't you be doing only this?
flag.setRetired(true);
flag.setRetireReason(retiredReason);
Context.getService(FlagsService.class).saveFlag(flag);

Reply:
Done since I am in the FlagServiceImpl class I just called saveFlag